lib/StringClient.js

```diff
@@ -18,14 +18,27 @@ var HttpClient = require('./HttpClient');
 function StringClient(options) {
     assert.object(options, 'options');
     assert.optionalObject(options.gzip, 'options.gzip');
+    assert.optionalObject(options.contentMd5, 'options.contentMd5');
+
+    if (options.contentMd5) {
+        assert.optionalArrayOfString(options.contentMd5.encodings,
+            'options.contentMd5.encodings');
+        assert.optionalBool(options.contentMd5.ignore,
+            'options.contentMd5.ignore');
+    }
 
     options.accept = options.accept || 'text/plain';
     options.name = options.name || 'StringClient';
     options.contentType =
         options.contentType || 'application/x-www-form-urlencoded';
 
     HttpClient.call(this, options);
+    this.contentMd5 = options.contentMd5 || {};
     this.gzip = options.gzip;
+
+    if (!this.contentMd5.encodings) {
+        this.contentMd5.encodings = [undefined];
```

Comment:
nit: Maybe add a comment like: "// the undefined value here is used to make node use the default encoding when computing the md5 hash"? Otherwise it's a bit confusing without reading the rest of the code way below.

Reply:
Okay - good idea.